enhancements/installer/kubevirt-ipi.md

```diff
@@ -0,0 +1,346 @@
+
+   With that the survey continues to the general cluster name, domain name, and 
+the rest of the non-KubeVirt specific question.
+
+2. Resource creation - terraform
+
+   Terraform uses kubernetes provider to create:
+
+    - [DataVolume CR][data-volumes] with RHCOS image
+
+         *Note:* In disconnected environment the user will need to provide a local image that the installer
+          can upload to the namespace.
+    - secrets for the Ignition configs of the VMs
+    - 1 bootstrap machine
+    - 3 masters
+    
+    Only on network option 1 - pods network:
+    - Services and routes for DNS and LB
+
+3. Bootstrap
+
+    The bootstrap VM has a huge Ignition config set using terraform as secrets and is visible
+as secrets on the infra OpenShift. KubeVirt boots that VM with that content as ConfigDrive and the 
+bootstraping begins when the `bootkube.service` systemd service starts.
+
+    This process described more thoroughly in the [installer overview document][https://github.com/OpenShift/installer/blob/37b99d8c9a3878bac7e8a94b6b0113fad6ffb77a/docs/user/overview.md#cluster-installation-process]
+
+4. Masters bootstrap
+
+    Master VMs are booting using a stub Ignition config that are waiting early in
+the Ignition service to load their Ignition config from a URL. That URL is the
+ `https://<internal-api-vip>/config/master` which is still not available until
+the **bootstrap** VM is exposing it. It takes few minutes till it does.
+
+    When the MachineConfigServer is available on the bootstrap, the masters pull their Ignition config 
+    and boot up joining the tenant cluster as masters and start scheduling pods.
+
+5. Workers bootstrap
+
+    After the masters and the control plane is up, we will scale the MachineSet to create workers
+    by the machine-api-operator.
+
+
+### Risks and Mitigations
+
+- Network
+
+    - Pods network option
+        -(OCP gap) The ports 22623/22624 that are used by MCS are blocked on the 
+        pods network and prevent from the nodes to pull Ignition config and updates.
+        - (KubeVirt gap) Interface binding - Currently the only supported binding on the pods
+        network is masquerade which means that all nodes are behind NAT, each VM
+        behind the NAT of his own pod.
+        - (OpenShift/KubeVirt gap) Static IP - OpenShift assumes that node's IP addresses are static,
+         but KubeVirt VMs change IP between restarts.
+        
+    - Secondary network option (MULTUS)
+    
+        - With this approach admin of the infra cluster will need to be involved in
+    the creation of each new tenant cluster since NADs need to be created and
+    probably also nmstate will need to be used to create the topology on the hosts.
+    In this proposal, we will assume that admin created the namespace and all network resources 
+    before running the installer, and the created networkName (NAD) will be the input for the installer.  
+
+    
+- Storage
```

Comment:
One of the other risks to call out is the **etcd** performance and latency requirements.  We've had issues where people deploy OCP clusters in virtual environments with insufficient hardware, and they have all sorts of problems installing the OCP clusters.  Worse, the cluster install may go fine, but the cluster goes unhealthy after a few days.  I'm not sure what the right answer is, but we should have a discussion about how we can make this easier to validate and troubleshoot.

Reply:
good point, I don't know how to solve it, especially when we are running on Baremetal and not on a public cloud